bot/exts/utilities/githubinfo.py

```diff
@@ -293,78 +311,142 @@ async def github_user_info(self, ctx: commands.Context, username: str) -> None:
 
         await ctx.send(embed=embed)
 
+    @tasks.loop(hours=24)
+    async def refresh_repos(self) -> None:
+        """Refresh self.repos with latest PyDis repos."""
+        self.repos, _ = await self.fetch_data(REPOSITORY_ENDPOINT.format(org="python-discord"))
+        log.info(f"Loaded {len(self.repos)} repos from Python Discord org into memory.")
+
     @github_group.command(name="repository", aliases=("repo",))
     async def github_repo_info(self, ctx: commands.Context, *repo: str) -> None:
         """
-        Fetches a repositories' GitHub information.
+        Fetches a repository's GitHub information.
 
         The repository should look like `user/reponame` or `user reponame`.
+        If it's not a stored repo or PyDis repo, it will fetch the most-starred repo
+        matching the search query from GitHub.
         """
-        repo = "/".join(repo)
-        if repo.count("/") != 1:
-            embed = discord.Embed(
-                title=random.choice(NEGATIVE_REPLIES),
-                description="The repository should look like `user/reponame` or `user reponame`.",
-                colour=Colours.soft_red
-            )
-
-            await ctx.send(embed=embed)
-            return
+        is_pydis = False
+        fetch_most_starred = False
+        repo_query = "/".join(repo)
+
+        # Determine type of repo
+        if repo_query.count("/") != 1:
+            if repo_query in self.stored_repos:
+                repo_query = self.stored_repos[repo_query]
+            else:
+                for each in self.repos:
+                    if repo_query == each["name"]:
+                        repo_query = each
+                        is_pydis = True
+                        break
+                else:
+                    fetch_most_starred = True
 
         async with ctx.typing():
-            repo_data, _ = await self.fetch_data(f"{GITHUB_API_URL}/repos/{quote(repo)}")
-
-            # There won't be a message key if this repo exists
-            if "message" in repo_data:
+            # Case 1: PyDis repo
+            if is_pydis:
+                repo_data = repo_query # repo_query already contains the matched repo
+
+            # Case 2: Not stored or PyDis, fetch most-starred matching repo
+            elif fetch_most_starred:
+                repo_data, _ = await self.fetch_data(FETCH_MOST_STARRED_ENDPOINT.format(name=quote(repo_query)))
+
+                if not repo_data["items"]:
+                    embed = discord.Embed(
+                        title=random.choice(NEGATIVE_REPLIES),
+                        description=f"No repositories found matching `{repo_query}`.",
+                        colour=Colours.soft_red
+                    )
+                    await ctx.send(embed=embed)
+                    return
+
+                repo_item = repo_data["items"][0]  # Top result
```

Comment:
```suggestion
                repo_data = repos["items"][0]  # Top result
                # Remove all the code below this since we no longer need it
```
If we do this, we can avoid duplicate code and just use the main parsing (the one that `is_pydis` and stored repos use). Kinda hard to explain, but I hope you know what I mean.

Do note that this case does not seem to check for a parent repo, so if there is a reasoning behind it, make sure to handle it